rust-lang/cargo
# [WIP] feat(gc): record workspace manifest and target dir in global cache tracker

### What does this PR try to resolve?
Fix #13136 
This PR aims to track of the following data in GC for further garbage collection use.
1. root-manifest path
2. target directory path

There will be three new table for this features.
1. worksapce_manifest_index table - index and record the root manifest path

id | workspace_manifest | timestamp
-- | -- | -- |
? | /foo/Cargo.toml | ?

2. target_dir_index table - index and record the target_dir path

id | target_dir | timestamp
-- | -- | -- |
? | /foo/target | ?

3. workspace_src table - combine the previous two table

id | worksapce_id | target_dir_id | timestamp
-- | -- | -- |
? | 1 | 1 | ?

### How should we test and review this PR?

1. Run compile (cargo run build or else)
2. inspect your root gc file (~/.cargo/.global-cache)
3. see if the three table is record correctly.
